azkaban-common/src/main/java/azkaban/executor/container/ContainerProxyUserCriteria.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.executor.container;
+
+import azkaban.Constants.ContainerizedDispatchManagerProperties;
+import azkaban.DispatchMethod;
+import azkaban.executor.ExecutableFlow;
+import azkaban.utils.Props;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Class for determining {@link DispatchMethod} based on deny list
+ */
+public class ContainerProxyUserCriteria {
+
+  private Set<String> denyList;
+
+  public ContainerProxyUserCriteria(final Props azkProps) {
+    this.denyList =
+        new HashSet<>(azkProps.getStringList(ContainerizedDispatchManagerProperties.CONTAINERIZED_PROXY_USER_DENYLIST));
+  }
+
+  @VisibleForTesting
+  public void updateDenyList(final Set<String> denyList) { this.denyList = denyList; }
```

Comment:
Do we need API options to update deny list without web server restart? If not, these methods are not required.

Reply:
removed